staging/src/k8s.io/code-generator/cmd/status-gen/status-generators/status.go

```diff
@@ -0,0 +1,442 @@
+		return false
+	}
+
+	if t.Kind != types.Struct {
+		return false
+	}
+
+	for _, currMember := range t.Members {
+		if currMember.Embedded && currMember.Name == "TypeMeta" {
+			return true
+		}
+	}
+
+	if t.Kind == types.Alias {
+		return isAPIType(t.Underlying)
+	}
+
+	return false
+}
+
+func (g *genAPIStatus) isOtherPackage(pkg string) bool {
+	if pkg == g.targetPackage {
+		return false
+	}
+	if strings.HasSuffix(pkg, "\""+g.targetPackage+"\"") {
+		return false
+	}
+	return true
+}
+
+func (g *genAPIStatus) Imports(c *generator.Context) (imports []string) {
+	importLines := []string{}
+	for _, singleImport := range g.imports.ImportLines() {
+		if g.isOtherPackage(singleImport) {
+			importLines = append(importLines, singleImport)
+		}
+	}
+	return importLines
+}
+
+func argsFromType(t *types.Type) (generator.Args, error) {
+	a := generator.Args{
+		"type": t,
+	}
+	introducedTag, introducedMajor, introducedMinor, err := extractIntroducedTag(t)
+	if err != nil {
+		return nil, err
+	}
+	a = a.
+		With("introduced", introducedTag.value).
+		With("introducedMajor", introducedMajor).
+		With("introducedMinor", introducedMinor)
+
+	// if someone intentionally override the deprecation release
+	if tagExists(deprecatedTagName, t) {
+		if tag, tagMajor, tagMinor, err := extractDeprecatedTag(t); err != nil {
+			return nil, err
+		} else {
+			a = a.With("deprecated", tag.value).
+				With("deprecatedMajor", tagMajor).
+				With("deprecatedMinor", tagMinor)
+		}
+
+	} else { // if not overridden, compute based on our policy
+		tagMajor := introducedMajor
+		tagMinor := introducedMinor + 3
+		a = a.With("deprecated", fmt.Sprintf("%d.%d", tagMajor, tagMinor)).
+			With("deprecatedMajor", tagMajor).
+			With("deprecatedMinor", tagMinor)
+	}
+
+	// if someone intentionally override the removed release
+	if tagExists(removedTagName, t) {
+		if tag, tagMajor, tagMinor, err := extractRemovedTag(t); err != nil {
+			return nil, err
+		} else {
+			a = a.With("removed", tag.value).
+				With("removedMajor", tagMajor).
+				With("removedMinor", tagMinor)
+		}
+
+	} else { // if not overridden, compute based on our policy
+		tagMajor := introducedMajor
+		tagMinor := introducedMinor + 6
+		a = a.With("removed", fmt.Sprintf("%d.%d", tagMajor, tagMinor)).
+			With("removedMajor", tagMajor).
+			With("removedMinor", tagMinor)
+	}
+
+	return a, nil
+}
+
+func (g *genAPIStatus) Init(c *generator.Context, w io.Writer) error {
+	return nil
+}
+
+func (g *genAPIStatus) GenerateType(c *generator.Context, t *types.Type, w io.Writer) error {
+	klog.Infof("Generating API Status for type %v", t)
+
+	sw := generator.NewSnippetWriter(w, c, "$", "$")
+	args, err := argsFromType(t)
+	if err != nil {
+		return err
+	}
+
+	if versionedMethodOrDie("Introduced", t) == nil {
+		sw.Do("// Introduced is an autogenerated function, returning the Kubernetes release in which the API struct was introduced and int versions of major and micro for comparison.\n", args)
```

Comment:
```suggestion
		sw.Do("// Introduced is an autogenerated function, returning the Kubernetes release in which the API struct was introduced and int versions of major and minor for comparison.\n", args)
```

Reply:
is it worth noting the comment format that controls this in a comment in these methods to save someone looking it up?

Reply:
done